#263 - Artworks for sale indicator

bexeGrace opened the pull request:
Closes #263

MAbdurrehman1 reviewed (changes requested):
Can we please do the following:

1. Add some space here: 
<img width="517" alt="image" src="https://github.com/thenewboston-developers/thenewboston-Frontend/assets/52558124/0c3f7c16-dcf1-4ca1-984c-1b6bfdaeacb1">

2. Make this a bit smaller as shown in the design in issue ticket
<img width="477" alt="image" src="https://github.com/thenewboston-developers/thenewboston-Frontend/assets/52558124/82c636b5-ea85-4396-9824-12014e35b668">

3. Combine these in 1 row, as shown in the design
<img width="506" alt="image" src="https://github.com/thenewboston-developers/thenewboston-Frontend/assets/52558124/7de1e2ad-d85a-4974-83e5-7dc692bfbec8">

4. Make the button label as same in the design
<img width="204" alt="image" src="https://github.com/thenewboston-developers/thenewboston-Frontend/assets/52558124/4284c873-167d-42c9-a2f9-328381e91222">

5. On the artworks on profile page, can we have some sort of differentiator/indicator that which artwork is in draft and which artwork is ready to be sold? I think we can use a badge here. (we already have a component for it)
<img width="773" alt="image" src="https://github.com/thenewboston-developers/thenewboston-Frontend/assets/52558124/7e64c6ee-93f1-47a9-8017-808fd3e60243">

6. Important one: If I have made an artwork ready to be sold on marketplace, and I edit it and save it as draft, it still shows up on the marketplace
<img width="387" alt="image" src="https://github.com/thenewboston-developers/thenewboston-Frontend/assets/52558124/f83118d1-957f-4ac9-aa87-8c3fe5782464">
<img width="539" alt="image" src="https://github.com/thenewboston-developers/thenewboston-Frontend/assets/52558124/7d714f5e-4c21-4dd1-bf8f-84eae5bb7ab5">
<img width="301" alt="image" src="https://github.com/thenewboston-developers/thenewboston-Frontend/assets/52558124/e80ddcaf-0693-46cf-bf86-1f7c1cba2cd3">

Solution on my mind: Currently, we do not have any flag in artwork model that can decide whether an artwork is in draft or not, since we are deciding that on whether the currency and amount are provided or not. I think this should be fine for now, but IF an artwork already has currency and amount then we don't show the toggle buttons to make it draft. OR we can set the currency and amount to null if user wants to draft the artwork again.

bexeGrace reviewed (commented):
Modified the required ui changes and added the badges on the artworks in profile page
![Screenshot (82)](https://github.com/thenewboston-developers/thenewboston-Frontend/assets/146959506/97891400-ee47-432a-9981-69f9a37190d4)
![Screenshot (83)](https://github.com/thenewboston-developers/thenewboston-Frontend/assets/146959506/6553d21a-304c-4164-aec1-5219a278486c)

bexeGrace commented:
removed the badges in the marketplace page and modified the UI in the following manner
![Screenshot (85)](https://github.com/thenewboston-developers/thenewboston-Frontend/assets/146959506/dc4a205c-87a8-4eb4-b830-cc0a1beaf9f9)
 manner

MAbdurrehman1 reviewed (changes requested):
Can we have these label in a single line? instead of going in next line.

<img width="445" alt="image" src="https://github.com/thenewboston-developers/thenewboston-Frontend/assets/52558124/8820b8ec-7d97-498b-960e-2c85aa8dc75a">

Whenever I edit an artwork from my profile, it just has 1 tab showing with 2 input fields, which I think is confusing for users.
<img width="538" alt="image" src="https://github.com/thenewboston-developers/thenewboston-Frontend/assets/52558124/e77673a7-3f49-4def-adf6-93a8dcf1beeb">
I think what we want to do here is: If an artwork is already in draft, then users still have the tabs options working but if an artwork is ready to be sold, and then we can hide the tabs options and show all fields (basically, we are not allowing users to make an active artwork as draft again). Does this makes sense?

The badges are showing on top of edit modals:
<img width="543" alt="image" src="https://github.com/thenewboston-developers/thenewboston-Frontend/assets/52558124/d8f64623-9121-469a-9915-736a70aabb5e">

bexeGrace commented on src/components/ArtworkCard/index.tsx:
```diff
@@ -1,23 +1,29 @@
 }
 
-const ArtworkCard: SFC<ArtworkCardProps> = ({artwork, className}) => {
+const ArtworkCard: SFC<ArtworkCardProps> = ({artwork, className, pageName}) => {
```
Okay

bexeGrace replied:
Modified it to be showBadge which accepts boolean value

bexeGrace commented on src/components/ArtworkCard/Styles.ts:
```diff
@@ -18,6 +17,7 @@ export const Container = styled.div`
   box-shadow: 0 3px 6px rgb(140 149 159 / 15%);
   overflow: hidden;
   transition: all 0.2s ease-in-out 0s;
+  position: relative; /* Ensure the container is positioned relative */
```
Alright

MAbdurrehman1 reviewed (approved):
👍
